src/lib/playerPlugins/login.js

```diff
@@ -40,78 +40,30 @@ function inject(serv,player)
     player.entity.position=toFixedPosition(player.spawnPoint);
   }
 
-  function spiral(arr)
-  {
-    var t=[];
-    spiralloop(arr,function(x,z){
-      t.push([x,z]);
-    });
-    return t;
-  }
-
-
-  function sendChunk(chunkX,chunkZ,column)
-  {
-    player._client.write('map_chunk', {
-      x: chunkX,
-      z: chunkZ,
-      groundUp: true,
-      bitMap: 0xffff,
-      chunkData: column.dump()
-    });
-    return Promise.resolve();
-  }
-
-  function sendChunksAroundPlayer(view)
-  {
-    player.lastPositionChunkUpdated=player.entity.position;
-    var playerChunkX=Math.floor(player.entity.position.x/16/32);
-    var playerChunkZ=Math.floor(player.entity.position.z/16/32);
-    return spiral([view*2,view*2])
-      .map(t => ({
-        chunkX:playerChunkX+t[0]-view,
-        chunkZ:playerChunkZ+t[1]-view
-      }))
-      .filter(({chunkX,chunkZ}) => {
-        var key=chunkX+","+chunkZ;
-        var loaded=player.loadedChunks[key];
-        if(!loaded) player.loadedChunks[key]=1;
-        return !loaded;
-      })
-      .reduce((acc,{chunkX,chunkZ})=>
-         acc
-          //.then(() => sleep(100))
-          .then(() => serv.world.getColumn(chunkX,chunkZ))
-          .then((column) => sendChunk(chunkX,chunkZ,column))
-      ,Promise.resolve());
-  }
-
   function sendMap()
   {
-    return sendChunksAroundPlayer(3);
+    return player.sendChunksAroundPlayer(3);
   }
 
   function sendRestMap()
   {
+    var initialChunks=player.sendChunksAroundPlayer(2);
     player.sendingChunks=true;
-    sendChunksAroundPlayer(player.view).then(() => player.sendingChunks=false);
+    player.sendChunksAroundPlayer(player.view).then(() => player.sendingChunks=false);
 
     player.on("positionChanged",function(){
       if(!player.sendingChunks && player.entity.position.distanceTo(player.lastPositionChunkUpdated)>16*32)
       {
         player.sendingChunks=true;
-        sendChunksAroundPlayer(player.view).then(() => player.sendingChunks=false);
+        player.sendChunksAroundPlayer(player.view).then(() => player.sendingChunks=false);
       }
     });
+    player.on("teleport", function() {
```

Comment:
No, that's not the way to do it. When a player teleport, its position should be changed (see updatePositions.js), which will then emit positionChanged, and send him the right chunks.